### DISCUSSION TITLE
How can I slide from different side by TransitioningContentControl?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#14361] [Q&A] [answered]
What I want to do is just like the video above:

https://github.com/AvaloniaUI/Avalonia/assets/30720036/6f76a057-1823-4e64-8d79-cba1d2a0c2b8

When clicking the setting button, it will slide from left to right.

When clicking the backward button, it will slide from right to left.

However, it seems PageSlide in TransitioningContentControl can only slide from left to right.

```
<TransitioningContentControl  Content="{Binding PageMode}">
				<TransitioningContentControl.PageTransition>
					<PageSlide Orientation="Horizontal" Duration="0:00:00.300" />
				</TransitioningContentControl.PageTransition>
				<TransitioningContentControl.DataTemplates>
					<TemplateSelectors:CurrentContentTemplateSelector>
						<DataTemplate x:Key="Setting" DataType="VM:PageEnum">
							<Local:SettingContent></Local:SettingContent>
						</DataTemplate>
						<DataTemplate x:Key="WifiSetting" DataType="VM:PageEnum">
							<Local:WifiSettingContent></Local:WifiSettingContent>
						</DataTemplate>
					</TemplateSelectors:CurrentContentTemplateSelector>
				</TransitioningContentControl.DataTemplates>
</TransitioningContentControl >
```

How should I achieve what I need? Thank you.

**Comment by kyuranger** [comment, marked as answer]
I found the solution. I should use the `carousel` in
[https://docs.avaloniaui.net/docs/reference/controls/carousel](https://docs.avaloniaui.net/docs/reference/controls/carousel)

This control is more suitable for my needs

**Reply by thevortexcloud** [reply]
For future reference you can sort of change the direction of a transition.

https://docs.avaloniaui.net/docs/guides/graphics-and-animation/page-transitions/page-transition-combinations

**Reply by kyuranger** [reply]
Thanks all the same.

**Comment by LITTOMA** [comment]
There's a `IsTransitionReversed` property added in #13003 . But it seems not available at the current release yet. You can copy the latest source code to use it (That's what I'm going to do).
